### PR TITLE
Add support for filename template

### PR DESCRIPTION
**Pull request by mAAdhaTTah**
Add setting for filename templates and pass it through the same templating logic as the link template.

### CODE REVIEW

**Comment by mAAdhaTTah** [comment]
Related to #5. I did this as I was working on expanding the templating logic but given we haven't actually agreed on the approach there, I didn't think it would be appropriate to do a bunch of work around it if you didn't want to manage that templating logic in this plugin. However, this seems both simple enough to implement as well as an acceptable new feature that I broke this off into its own PR.

~~Not yet tested in my own vault.~~ Edit: Tested and confirmed working.

File: src/main.ts [hunk 1]
```diff
@@ -199,6 +209,18 @@ class LinkStowrSettingTab extends PluginSettingTab {
           }),
       );
 
+    new Setting(containerEl)
+      .setName('Link filename template')
```
Comment (joelseq) [inline src/main.ts]: nit: Can we call this just `Filename template` instead?

File: src/main.ts [hunk 2]
```diff
@@ -199,6 +209,18 @@ class LinkStowrSettingTab extends PluginSettingTab {
...
+      .setDesc('Enter template for filenames generated by Linkstower')
```
Comment (joelseq) [inline src/main.ts]: typo:
```suggestion
      .setDesc('Enter template for filenames generated by LinkStowr')
```

**Review by joelseq — COMMENTED** [review]
Sorry for the delay in reviewing this! Mostly LGTM but just left a couple of comments. Will approve once those are addressed!

**Comment by mAAdhaTTah** [comment]
Fixed!

**Review by joelseq — APPROVED** [review]

**Comment by joelseq** [comment]
Thanks for making the changes and for the PR @mAAdhaTTah!

**Pull request merged**